### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ChrisTerBeke**
This is the first PR as part of the .curapackage implementation. This PR refactors OPC out to a new class and makes UFP an inherited class of that. It also contains a few bugfixes and codestyle fixes.

### CODE REVIEW

File: Charon/VirtualFile.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-from .FileInterface import FileInterface #The interface we're implementing.
-from .OpenMode import OpenMode #To open local files with the selected open mode.
-
-#The supported file types.
-from .filetypes.UltimakerFormatPackage import UltimakerFormatPackage
-from .filetypes.GCodeFile import GCodeFile
-from .filetypes.GCodeGzFile import GCodeGzFile
 
 extension_to_mime = {
-    ".ufp": "application/x-ufp",
```
Comment (daid) [inline Charon/VirtualFile.py]: Removing the ufp format from the VirtualFile causes the printer no longer being able to read ufp files. Unintentional regression?

Comment (ChrisTerBeke) [reply Charon/VirtualFile.py]: That should be fixed indeed. I'll do that now.

File: Charon/VirtualFile.py [hunk 2]
```diff
@@ -1,23 +1,26 @@
 # Copyright (c) 2018 Ultimaker B.V.
 # libCharon is released under the terms of the LGPLv3 or higher.
+import os
 
-import os.path #To get the file extension from a path.
+from Charon.FileInterface import FileInterface  # The interface we're implementing.
+from Charon.OpenMode import OpenMode  #T o open local files with the selected open mode.
```
Comment (nallath) [inline Charon/VirtualFile.py]: Nitpicking: should be # To open ;)

File: Charon/VirtualFile.py [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
-from .FileInterface import FileInterface #The interface we're implementing.
-from .OpenMode import OpenMode #To open local files with the selected open mode.
-
-#The supported file types.
-from .filetypes.UltimakerFormatPackage import UltimakerFormatPackage
-from .filetypes.GCodeFile import GCodeFile
-from .filetypes.GCodeGzFile import GCodeGzFile
 
 extension_to_mime = {
+    ".opc": "application/x-opc",
```
Comment (nallath) [inline Charon/VirtualFile.py]: I'm not even sure that there is such a thing as a .opc file. I've not been able to find any reference to it what so ever. So i think we should just keep the OPC support as a sort of interface class (as a raw OPC file doesn't make any sense. It needs to have an actual implementation.)

**Pull request merged**